ENH: Expose custom vertex params in Aperture

TEParsons opened the pull request.

tomwallis committed:
Merge remote-tracking branch 'psychopy/master'

tomwallis committed:
Merge remote-tracking branch 'psychopy/master'

tomwallis committed:
Merge remote-tracking branch 'psychopy/master'

tomwallis committed:
Merge remote-tracking branch 'psychopy/master'

tomwallis committed:
Merge remote-tracking branch 'psychopy/master'

tomwallis committed:
Merge remote-tracking branch 'psychopy/master'

tomwallis committed:
Merge remote-tracking branch 'psychopy/master'

TEParsons committed:
DOC: Add documentation for contributing to the test suite

TEParsons committed:
ENH: Expose custom vertex params in Aperture

TEParsons committed:
DOC: Update ApertureStim doc string

peircej commented:
Task linked: [CU-nqd0ya Update Builder component docs to describe new exposed params](https://app.clickup.com/t/nqd0ya)

peircej commented:
Task linked: [CU-nqcypn Aperture component updates](https://app.clickup.com/t/nqcypn)

guillermoaguilar committed:
BF: Monitor Center hangs when calling Plot Gamma after checking monitor linearization

TEParsons committed:
DOC: Sphinx-friendly underlines

TEParsons committed:
FF: Mark shape as non-direct

mdcutone committed:
BF: Fixed warning raised in `ShapeStim` caused by doing comparison with empty sequence.

mdcutone committed:
RF: `tostring` -> `tobytes` for mask data in `RadialStim`

mdcutone committed:
RF: Removed unused variable `fromFile`

mdcutone committed:
RF: Check argument value type using `isinstance` in `RadialStim`

mdcutone committed:
RF: Check argument value type using `isinstance` in `SimpleImageStim`

mdcutone committed:
RF: Use `tobytes` instead of `tostring` in `SimpleImageStim._updateImageStr`

mdcutone committed:
RF: Remove deprecated `np.float` in `helpers.py`

mdcutone committed:
RF: Remove deprecated `np.long` and `np.int` in `helpers.py`

mdcutone committed:
RF: Added `np.float32` as immutable type for verts

mdcutone committed:
FF: Check if value is a string before making the comparison with colour names in the list. Check if sequence for vertex data is larger than zero.

mdcutone committed:
RF: Remove deprecated behaviour by using `tobytes` in `imagetools.array2image`

mdcutone committed:
RF: Use `tobytes` and `frombuffer` for `imagetools` functions.

mdcutone committed:
RF; Use `asarray` instead of `asmatrix` for `colorspacetools.cielab2rgb()`.

mdcutone committed:
RF; Use `asarray` instead of `asmatrix` for `calibTools.makeXYZ2RGB()`.

mdcutone committed:
BF: Catch `AttributeError` and ignore when shutting down if `com` is `None`

peircej committed:
Merge pull request #4221 from mdcutone/release-crs-fix

BF: Catch `AttributeError` in `BitsSharp.__del__`

peircej committed:
Merge pull request #4220 from mdcutone/release-warning-fixes

RF/BF: Fix a bunch of warnings cropping up during builds

mdcutone committed:
FF: Make sure the math works when using arrays instead of matrices

mdcutone committed:
DOCS: Fixed some formatting and typos.

mdcutone committed:
DOCS: Added example.

mdcutone committed:
RF: Bump up threshold for timing in VMs to prevent failures on MacOS.

mdcutone committed:
Merge remote-tracking branch 'remote/release' into release-warning-fixes

peircej committed:
Merge pull request #4222 from mdcutone/release-warning-fixes

FF/TEST: Fixed issues around using arrays in `MakeXYZ2RGB` and modified frame timing threshold for VMs

guillermoaguilar committed:
BF: Monitor Center hangs when calling Plot Gamma after checking monitor linearization - commit 2

TEParsons committed:
ENH: Add util.randchoice to JS transpiler

peircej commented:
Task linked: [CU-nqcz64 Expose "shape"](https://app.clickup.com/t/nqcz64)

peircej committed:
Merge pull request #4218 from guillermoaguilar/BF-plotgamma-hangs

BF: Monitor Center hangs when calling "Plot gamma" after linearization

mdcutone committed:
BF: Check for empty lists when specifying vertices in `BaseShapeStim`

mdcutone committed:
FF: Just revert back and eat the warning for now

mdcutone committed:
BF: Fixed unhandled error when using clipboard hotkeys.

mdcutone committed:
BF: Fixed calltip popup location.

peircej committed:
Merge pull request #4240 from mdcutone/release-coder-clipboard-bf

BF: Fixed unhandled error when using clipboard hotkeys in coder

peircej committed:
Merge pull request #4231 from mdcutone/release-shape-vert-bf

BF: Check for empty lists when specifying vertices in `BaseShapeStim`

TEParsons committed:
BF: Pavlovia Commit dlg Okay button was on wrong side

RebeccaHirst committed:
DOCS: maintenence of screenshots and credits

isolver committed:
BF: directly using iohub.client.ioHubConnection

Client script use of client.ioHubConnection (which is not longer recommended) was failing as of 2021.2.x if using a Tobii, GazePoint, or Mousegaze tracker. This fixes that regression.

TEParsons committed:
BF: Update Pavlovia project root on sync, fixes #4227

RebeccaHirst committed:
DOCS: coder info (update screenshots, correct formatting errors, add more links to resources)

RebeccaHirst committed:
DOCS: update overview page on about page

peircej committed:
Merge pull request #4249 from isolver/hotfix-using-client.ioHubConnection

BF: directly using iohub.client.ioHubConnection

RebeccaHirst committed:
correct underline length

RebeccaHirst committed:
ENH(?): add toLowerCase() method to transpiler to translate the lower() method

RebeccaHirst committed:
ENH(?): add JS version of a.extend() method

RebeccaHirst committed:
BF: add JS version of a.upper() method

peircej committed:
Merge pull request #4247 from TEParsons/_okPavloviaCommit

BF: Pavlovia Commit dlg Okay button was on wrong side

TEParsons committed:
BF: Still write STARTED/FINISHED code for Mouse when no response gathered

TEParsons committed:
TEST: Add test for transpiler substitutions

peircej committed:
Merge pull request #4256 from RebeccaHirst/_transpilerUpperCase

BF: Transpiler add JS version of a.upper() method

peircej committed:
Merge pull request #4260 from TEParsons/_mouseFinished

BF: Still write STARTED/FINISHED code for Mouse when no response gathered, fixes #4246

peircej committed:
Merge pull request #4255 from RebeccaHirst/_transpilerAddConcat

BF: Transpiler add JS version of a.extend() method

peircej committed:
Merge pull request #4204 from TEParsons/_contribTestSuite

DOC: Add documentation for contributing to the test suite

peircej committed:
Merge pull request #4253 from RebeccaHirst/_updateOverviewPage

DOCS: update overview page on about page

peircej committed:
Merge pull request #4254 from RebeccaHirst/_addToLowerCase

BF: Transpiler add toLowerCase() method to transpiler

peircej committed:
Merge pull request #4251 from TEParsons/_pavloviaRootChange

BF: Update Pavlovia project root on sync, fixes #4227

TEParsons committed:
Merge remote-tracking branch 'upstream/release' into _substitutionTest

hsogo committed:
BF: Coder's "Find" doesn't work properly when text includes non-ASCII characters

datalowe committed:
make slider component's corresponding mouse object be connected to the same window as the slider itself.

peircej committed:
Merge pull request #4271 from datalowe/release

Link slider component's mouse to the correct window

peircej committed:
Merge pull request #4263 from TEParsons/_substitutionTest

TEST: Test JS substitutions

peircej committed:
Merge pull request #4272 from hsogo/fix-coder-find

BF: Coder's "Find" fails when non-ASCII characters are included

ClionaOD committed:
BF: fixed reset method to make next frame 0.0. Currently setting to None throws an error on repetition of videos (#4267)

peircej committed:
Merge pull request #4248 from RebeccaHirst/_updateScreenShots

DOCS: maintenance of screenshots and credits

peircej committed:
Merge pull request #4252 from RebeccaHirst/_updateCoderDocs

DOCS: Update coder docs

TEParsons committed:
Merge pull request #4228 from TEParsons/_jsRandchoice

ENH: Add util.randchoice to JS transpiler

Merging in as companion PR ( https://github.com/psychopy/psychojs/pull/457 ) is pulled into JS

tomwallis committed:
Merge branch 'psychopy:release' into master

tomwallis committed:
Add examples of showing images

There are a few gotchas that happen when showing numpy arrays
as images. This shows one way to get an image on the screen.

sappelhoff committed:
DOC: update SR Research forum links (#4292)

TEParsons committed:
BF: Add Beta sticker back onto Button (#4291)

mdcutone committed:
BF: Add missing requirements for transcription libraries. (#4276)

tomwallis committed:
Merge branch 'psychopy:release' into showImage_doc

tomwallis committed:
DOC: remove skimage dependency in showImages

tomwallis committed:
DOC: Add more information on image to docstring.

tomwallis committed:
DOC: Add more information on image to docstring.

tomwallis committed:
DOC: fix accidental Black reformatting

tomwallis committed:
DOC: Fix swapped width and height

peircej committed:
Merge pull request #4296 from tomwallis/showImage_doc

**DOC**: Showing a numpy array as an image with ImageStim

TEParsons committed:
BF: Texture resolution needs to be code, not num, so it stays an int (#4266)

peircej committed:
TESTS: fix macOS test failure to compile pocketsphinx

peircej committed:
TESTS: try to fix pocketsphinx install with brew install openal-soft

TEParsons committed:
BF: Include font name in warning when not found

peircej committed:
Merge pull request #4311 from TEParsons/_fontSearch

BF: Include font name in warning when not found

TEParsons committed:
BF: Polygon should be white by default, not None (#4306)

TEParsons committed:
Merge remote-tracking branch 'upstream/dev' into _apertureComponent

TEParsons committed:
RF: Ignore add venvs

Commit:
BF: Fixed using wrong ID for copy/open in runner

peircej committed:
Merge pull request #4317 from mdcutone/release-output-hotfix

BF: Fixed using wrong ID for copy/open in runner

peircej committed:
Merge pull request #4316 from TEParsons/_ignoreVenv

RF: Ignore add venvs

TEParsons committed:
Merge remote-tracking branch 'upstream/release' into _apertureComponent

# Conflicts:
#	setup.cfg

Commit:
Merge remote-tracking branch 'upstream/dev' into _apertureComponent

TEParsons committed:
Merge remote-tracking branch 'upstream/dev' into _apertureComponent

# Conflicts:
#	setup.cfg

TEParsons committed:
Merge remote-tracking branch 'origin/_apertureComponent' into _apertureComponent
